### ISSUE TITLE
Pointnet dependencies

### ISSUE THREAD

**Issue by kostasfnrs** [neu-vi/HOI-Diff#20]
Hello, thank you for your research. I have some trouble when I am trying to install pointnet in order to train the affordance prediction module. How did you manage to use pointnet for your work? The official repo of pointnet is currently unmaintained. Thanks for your time

**Comment by fivezerojun** [comment]
Hello, I also encountered trouble installing PointNet. I couldn't successfully configure it while using the HOI-Diff conda environment. Did you solve the problem? Thanks for your time.

**Comment by kostasfnrs** [comment]
Hi @fivezerojun. Start building the Pointnet first and then the hoi-diff on top of it. You can do the following:
First create an anaconda environment using Python 2.7:
conda create --name pointnet2_env python=2.7

Then install these versions of torch and torch vision with cuda11.7
pip install torch==1.13.0+cu117 torchvision==0.14.0+cu117 --extra-index-url https://download.pytorch.org/whl/cu117
Ensure that the environment paths are correctly set to the CUDA toolkit you intend to use. If necessary, adjust your environment variables:
export PATH=/path/to/conda/envs/myenv/bin:$PATH (set your conda path)
export LD_LIBRARY_PATH=/path/to/conda/envs/myenv/lib:$LD_LIBRARY_PATH (set your conda path)


Install the correct g++ version supported by cuda11.7
conda install -c conda-forge gxx_linux-64=9.3.0

Set the Environment Variables:
export CXX=/path/to/conda/envs/pointnet2_env/bin/x86_64-conda-linux-gnu-g++
export CC=/path/to/conda/envs/pointnet2_env/bin/x86_64-conda-linux-gnu-gcc

Ensure the correct version of g++ is being used by your anaconda environment
First, identify the path to the g++ provided by your Conda environment:
which x86_64-conda-linux-gnu-g++
Export the CXX variable with the correct path:
export CXX=$(which x86_64-conda-linux-gnu-g++)
Verify the CXX version:
$CXX --version

Repeat the process for the CC environment variable:
export CC=$(which x86_64-conda-linux-gnu-gcc)
Verify the CC version:
$CC --version

If after all these you get an error indicating that the build process is failing due to a missing crypt.h header file, which is a part of the libxcrypt development package, do the following

conda install -c conda-forge libxcrypt

Make sure your environment variables are set correctly to point to the conda-installed compilers and include directories.
export CC=~/miniconda3/envs/pointnet2_env/bin/x86_64-conda-linux-gnu-gcc
export CXX=~/miniconda3/envs/pointnet2_env/bin/x86_64-conda-linux-gnu-g++
export C_INCLUDE_PATH=~/miniconda3/envs/pointnet2_env/include
export CPLUS_INCLUDE_PATH=~/miniconda3/envs/pointnet2_env/include


After installing the required package and setting the environment variables, try to install pointnet2-ops again:
pip install pointnet2-ops

You also need to install these required pip packages for hoi-diff:
pip install chumpy==0.70 \
  freetype-py==2.4.0 \
  future==1.0.0 \
  imageio==2.34.0 \
  networkx==3.1 \
  numpy==1.23.1 \
  pyglet==2.0.14 \
  pyopengl==3.1.0 \
  pyrender==0.1.45 \
  pyyaml==6.0.1 \
  scipy==1.10.1 \
  six==1.16.0 \
  smplx==0.1.28 \
  torchgeometry==0.1.2 \
  tqdm==4.66.2 \
  trimesh==4.1.8

**Comment by zhenkirito123** [comment]
Thanks for the detailed guidance! However, Python 2.7 doesn't work in my case. I tried to use Python 3.7 and it works :)

**Comment by hosamanivinayak5** [comment]
Hello  can you please guide me how to setup the same in windows .
My dependencies are:
 -python-3.7
 - blas=1.0=mkl
  - brotli=1.0.9=h827c3e9_9
  - brotli-bin=1.0.9=h827c3e9_9
  - brotlipy=0.7.0=py37h2bbff1b_1003
  - ca-certificates=2025.2.25=haa95532_0
  - catalogue=2.0.7=py37haa95532_0
  - certifi=2022.12.7=py37haa95532_0
  - cffi=1.15.1=py37h2bbff1b_3
  - charset-normalizer=2.0.4=pyhd3eb1b0_0
  - click=8.0.4=py37haa95532_0
  - colorama=0.4.6=py37haa95532_0
  - cryptography=39.0.1=py37h21b164f_0
  - cudatoolkit=11.0.221=h74a9793_0
  - cycler=0.11.0=pyhd3eb1b0_0
  - cymem=2.0.6=py37hd77b12b_0
  - cython-blis=0.7.7=py37h080aedc_0
  - fftw=3.3.9=h2bbff1b_2
  - fonttools=4.25.0=pyhd3eb1b0_0
  - freetype=2.13.3=h0620614_0
  - h5py=3.7.0=py37h3de5c98_0
  - hdf5=1.10.6=h1756f20_1
  - icc_rt=2022.1.0=h6049295_2
  - icu=58.2=ha925a31_3
  - idna=3.4=py37haa95532_0
  - importlib-metadata=4.11.3=py37haa95532_0
  - intel-openmp=2021.4.0=haa95532_3556
  - jinja2=3.1.2=py37haa95532_0
  - joblib=1.1.1=py37haa95532_0
  - jpeg=9b=hb83a4c4_2
  - kiwisolver=1.4.4=py37hd77b12b_0
  - langcodes=3.3.0=pyhd3eb1b0_0
  - lerc=4.0.0=h5da7b33_0
  - libbrotlicommon=1.0.9=h827c3e9_9
  - libbrotlidec=1.0.9=h827c3e9_9
  - libbrotlienc=1.0.9=h827c3e9_9
  - libdeflate=1.22=h5bf469e_0
  - libpng=1.6.39=h8cc25b3_0
  - libtiff=4.5.0=h8a3f274_0
  - libuv=1.48.0=h827c3e9_0
  - libwebp=1.2.4=h2bbff1b_0
  - libwebp-base=1.2.4=h2bbff1b_1
  - lz4-c=1.9.4=h2bbff1b_1
  - markupsafe=2.1.1=py37h2bbff1b_0
  - matplotlib=3.5.3=py37haa95532_0
  - matplotlib-base=3.5.3=py37hd77b12b_0
  - mkl=2021.4.0=haa95532_640
  - mkl-service=2.4.0=py37h2bbff1b_0
  - mkl_fft=1.3.1=py37h277e83a_0
  - mkl_random=1.2.2=py37hf11a4ad_0
  - munkres=1.1.4=py_0
  - murmurhash=1.0.7=py37hd77b12b_0
  - ninja=1.12.1=haa95532_0
  - ninja-base=1.12.1=h4157e71_0
  - numpy=1.21.5=py37h7a0a035_3
  - numpy-base=1.21.5=py37hca35cd5_3
  - openssl=1.1.1w=h2bbff1b_0
  - packaging=22.0=py37haa95532_0
  - pathy=0.10.1=py37haa95532_0
  - pillow=9.2.0=py37hdc2b20a_1
  - pip=22.3.1=py37haa95532_0
  - preshed=3.0.6=py37h6c2663c_0
  - pycparser=2.21=pyhd3eb1b0_0
  - pyopenssl=23.0.0=py37haa95532_0
  - pyparsing=3.0.9=py37haa95532_0
  - pyqt=5.9.2=py37hd77b12b_6
  - pysocks=1.7.1=py37_1
  - python=3.7.16=h6244533_0
  - python-dateutil=2.8.2=pyhd3eb1b0_0
  - pytorch=1.7.1=py3.7_cuda110_cudnn8_0
  - qt=5.9.7=vc14h73c81de_0
  - requests=2.28.1=py37haa95532_0
  - scikit-learn=1.0.2=py37hf11a4ad_1
  - scipy=1.7.3=py37h7a0a035_2
  - setuptools=65.6.3=py37haa95532_0
  - shellingham=1.5.0=py37haa95532_0
  - sip=4.19.13=py37hd77b12b_0
  - six=1.16.0=pyhd3eb1b0_1
  - smart_open=5.2.1=py37haa95532_0
  - spacy=3.3.1=py37h0f1f7c2_0
  - spacy-legacy=3.0.9=py37haa95532_0
  - spacy-loggers=1.0.1=pyhd3eb1b0_0
  - sqlite=3.45.3=h2bbff1b_0
  - srsly=2.4.3=py37hd77b12b_0
  - thinc=8.0.15=py37h757b272_0
  - threadpoolctl=2.2.0=pyh0d69192_0
  - tk=8.6.14=h0416ee5_0
  - torchaudio=0.7.2=py37
  - torchvision=0.8.2=py37_cu110
  - tornado=6.2=py37h2bbff1b_0
  - tqdm=4.64.1=py37haa95532_0
  - trimesh=4.5.3=pyhd8ed1ab_0
  - typer=0.4.1=py37haa95532_0
  - typing-extensions=3.10.0.2=hd3eb1b0_0
  - typing_extensions=3.10.0.2=pyh06a4308_0
  - urllib3=1.26.14=py37haa95532_0
  - vc=14.42=haa95532_5
  - vs2015_runtime=14.42.34433=hbfb602d_5
  - wasabi=0.9.1=py37haa95532_0
  - wheel=0.38.4=py37haa95532_0
  - win_inet_pton=1.1.0=py37haa95532_0
  - wincertstore=0.2=py37haa95532_2
  - xz=5.6.4=h4754444_1
  - zipp=3.11.0=py37haa95532_0
  - zlib=1.2.13=h8cc25b3_1
  - zstd=1.5.6=h8880b57_0
  - pip:
      - absl-py==2.1.0
      - beautifulsoup4==4.13.4
      - blobfile
      - cachetools==5.5.2
      - chumpy==0.70
      - clip
      - en-core-web-sm==3.3.0
      - filelock==3.12.2
      - ftfy==6.1.1
      - future==1.0.0
      - gdown==4.7.3
      - google-auth==2.40.1
      - google-auth-oauthlib==0.4.6
      - grpcio==1.62.3
      - hydra-core==0.11.3
      - lmdb==1.6.2
      - lxml==4.9.4
      - markdown==3.4.4
      - msgpack==1.0.5
      - msgpack-numpy==0.4.8
      - oauthlib==3.2.2
      - omegaconf==1.4.1
      - pointnet2==3.0.0
      - protobuf==3.20.3
      - pyasn1==0.5.1
      - pyasn1-modules==0.3.0
      - pycryptodomex==3.22.0
      - pydantic==1.8.2
      - pytorch-lightning==0.7.1
      - pyyaml==6.0.1
      - regex==2024.4.16
      - requests-oauthlib==2.0.0
      - rsa==4.9.1
      - smplx==0.1.28
      - soupsieve==2.4.1
      - tensorboard==2.11.2
      - tensorboard-data-server==0.6.1
      - tensorboard-plugin-wit==1.8.1
      - wcwidth==0.2.13
      - werkzeug==2.2.3

**Comment by SanjanaSK03** [comment]
> Hello can you please guide me how to setup the same in windows . My dependencies are: -python-3.7
> 
> * blas=1.0=mkl
> * brotli=1.0.9=h827c3e9_9
> * brotli-bin=1.0.9=h827c3e9_9
> * brotlipy=0.7.0=py37h2bbff1b_1003
> * ca-certificates=2025.2.25=haa95532_0
> * catalogue=2.0.7=py37haa95532_0
> * certifi=2022.12.7=py37haa95532_0
> * cffi=1.15.1=py37h2bbff1b_3
> * charset-normalizer=2.0.4=pyhd3eb1b0_0
> * click=8.0.4=py37haa95532_0
> * colorama=0.4.6=py37haa95532_0
> * cryptography=39.0.1=py37h21b164f_0
> * cudatoolkit=11.0.221=h74a9793_0
> * cycler=0.11.0=pyhd3eb1b0_0
> * cymem=2.0.6=py37hd77b12b_0
> * cython-blis=0.7.7=py37h080aedc_0
> * fftw=3.3.9=h2bbff1b_2
> * fonttools=4.25.0=pyhd3eb1b0_0
> * freetype=2.13.3=h0620614_0
> * h5py=3.7.0=py37h3de5c98_0
> * hdf5=1.10.6=h1756f20_1
> * icc_rt=2022.1.0=h6049295_2
> * icu=58.2=ha925a31_3
> * idna=3.4=py37haa95532_0
> * importlib-metadata=4.11.3=py37haa95532_0
> * intel-openmp=2021.4.0=haa95532_3556
> * jinja2=3.1.2=py37haa95532_0
> * joblib=1.1.1=py37haa95532_0
> * jpeg=9b=hb83a4c4_2
> * kiwisolver=1.4.4=py37hd77b12b_0
> * langcodes=3.3.0=pyhd3eb1b0_0
> * lerc=4.0.0=h5da7b33_0
> * libbrotlicommon=1.0.9=h827c3e9_9
> * libbrotlidec=1.0.9=h827c3e9_9
> * libbrotlienc=1.0.9=h827c3e9_9
> * libdeflate=1.22=h5bf469e_0
> * libpng=1.6.39=h8cc25b3_0
> * libtiff=4.5.0=h8a3f274_0
> * libuv=1.48.0=h827c3e9_0
> * libwebp=1.2.4=h2bbff1b_0
> * libwebp-base=1.2.4=h2bbff1b_1
> * lz4-c=1.9.4=h2bbff1b_1
> * markupsafe=2.1.1=py37h2bbff1b_0
> * matplotlib=3.5.3=py37haa95532_0
> * matplotlib-base=3.5.3=py37hd77b12b_0
> * mkl=2021.4.0=haa95532_640
> * mkl-service=2.4.0=py37h2bbff1b_0
> * mkl_fft=1.3.1=py37h277e83a_0
> * mkl_random=1.2.2=py37hf11a4ad_0
> * munkres=1.1.4=py_0
> * murmurhash=1.0.7=py37hd77b12b_0
> * ninja=1.12.1=haa95532_0
> * ninja-base=1.12.1=h4157e71_0
> * numpy=1.21.5=py37h7a0a035_3
> * numpy-base=1.21.5=py37hca35cd5_3
> * openssl=1.1.1w=h2bbff1b_0
> * packaging=22.0=py37haa95532_0
> * pathy=0.10.1=py37haa95532_0
> * pillow=9.2.0=py37hdc2b20a_1
> * pip=22.3.1=py37haa95532_0
> * preshed=3.0.6=py37h6c2663c_0
> * pycparser=2.21=pyhd3eb1b0_0
> * pyopenssl=23.0.0=py37haa95532_0
> * pyparsing=3.0.9=py37haa95532_0
> * pyqt=5.9.2=py37hd77b12b_6
> * pysocks=1.7.1=py37_1
> * python=3.7.16=h6244533_0
> * python-dateutil=2.8.2=pyhd3eb1b0_0
> * pytorch=1.7.1=py3.7_cuda110_cudnn8_0
> * qt=5.9.7=vc14h73c81de_0
> * requests=2.28.1=py37haa95532_0
> * scikit-learn=1.0.2=py37hf11a4ad_1
> * scipy=1.7.3=py37h7a0a035_2
> * setuptools=65.6.3=py37haa95532_0
> * shellingham=1.5.0=py37haa95532_0
> * sip=4.19.13=py37hd77b12b_0
> * six=1.16.0=pyhd3eb1b0_1
> * smart_open=5.2.1=py37haa95532_0
> * spacy=3.3.1=py37h0f1f7c2_0
> * spacy-legacy=3.0.9=py37haa95532_0
> * spacy-loggers=1.0.1=pyhd3eb1b0_0
> * sqlite=3.45.3=h2bbff1b_0
> * srsly=2.4.3=py37hd77b12b_0
> * thinc=8.0.15=py37h757b272_0
> * threadpoolctl=2.2.0=pyh0d69192_0
> * tk=8.6.14=h0416ee5_0
> * torchaudio=0.7.2=py37
> * torchvision=0.8.2=py37_cu110
> * tornado=6.2=py37h2bbff1b_0
> * tqdm=4.64.1=py37haa95532_0
> * trimesh=4.5.3=pyhd8ed1ab_0
> * typer=0.4.1=py37haa95532_0
> * typing-extensions=3.10.0.2=hd3eb1b0_0
> * typing_extensions=3.10.0.2=pyh06a4308_0
> * urllib3=1.26.14=py37haa95532_0
> * vc=14.42=haa95532_5
> * vs2015_runtime=14.42.34433=hbfb602d_5
> * wasabi=0.9.1=py37haa95532_0
> * wheel=0.38.4=py37haa95532_0
> * win_inet_pton=1.1.0=py37haa95532_0
> * wincertstore=0.2=py37haa95532_2
> * xz=5.6.4=h4754444_1
> * zipp=3.11.0=py37haa95532_0
> * zlib=1.2.13=h8cc25b3_1
> * zstd=1.5.6=h8880b57_0
> * pip:
>   
>   * absl-py==2.1.0
>   * beautifulsoup4==4.13.4
>   * blobfile
>   * cachetools==5.5.2
>   * chumpy==0.70
>   * clip
>   * en-core-web-sm==3.3.0
>   * filelock==3.12.2
>   * ftfy==6.1.1
>   * future==1.0.0
>   * gdown==4.7.3
>   * google-auth==2.40.1
>   * google-auth-oauthlib==0.4.6
>   * grpcio==1.62.3
>   * hydra-core==0.11.3
>   * lmdb==1.6.2
>   * lxml==4.9.4
>   * markdown==3.4.4
>   * msgpack==1.0.5
>   * msgpack-numpy==0.4.8
>   * oauthlib==3.2.2
>   * omegaconf==1.4.1
>   * pointnet2==3.0.0
>   * protobuf==3.20.3
>   * pyasn1==0.5.1
>   * pyasn1-modules==0.3.0
>   * pycryptodomex==3.22.0
>   * pydantic==1.8.2
>   * pytorch-lightning==0.7.1
>   * pyyaml==6.0.1
>   * regex==2024.4.16
>   * requests-oauthlib==2.0.0
>   * rsa==4.9.1
>   * smplx==0.1.28
>   * soupsieve==2.4.1
>   * tensorboard==2.11.2
>   * tensorboard-data-server==0.6.1
>   * tensorboard-plugin-wit==1.8.1
>   * wcwidth==0.2.13
>   * werkzeug==2.2.3

Encountering same issues, please guide me to set up Pointnet in Windows
